src/utils/diff/index.ts

```diff
@@ -0,0 +1,66 @@
+import * as jsdiff from 'diff';
+import { colorLines } from './colors';
+import { type, emptyRepresentation, cleanUp } from './util';
+import { INDENT } from './const';
+
+
+function unifiedDiff (actual: string, expected: string): string {
+    const msg = jsdiff.createPatch('string', actual, expected);
+    const lines = msg.split('\n').splice(5);
+
+    return (
+        colorLines('diff added', '+ expected') +
+        ' ' +
+        colorLines('diff removed', '- actual') +
+        '\n\n' +
+        lines
+            .map(cleanUp)
+            .filter((line: any) => typeof line !== 'undefined' && line !== null)
+            .join('\n')
+    );
+}
+
+export function generateDiff (actual: string, expected: string): string {
+    try {
+        return unifiedDiff(actual, expected);
+    }
+    catch (err) {
+        const msg =
+            colorLines('diff removed', 'Failed to generate diff') +
+            '\n';
+
+        return msg;
+    }
+}
+
+export function stringify (value: any): string {
+    const typeHint = type(value);
+
+    if (['object', 'array'].indexOf(typeHint) !== -1) {
```

Comment:
Please use constants for literals.

Reply:
I feel that this function can be simplified by using `util.inspect` instead of `JSON.stringify`